convert/jenkinsjson/convert.go

```diff
@@ -163,35 +163,40 @@ func recursiveParseJsonToStages(jsonNode *jenkinsjson.Node, dst *harness.Pipelin
 }
 
 func collectStagesWithID(jsonNode *jenkinsjson.Node, processedTools *ProcessedTools, stepGroupWithId *[]StepGroupWithID, variables map[string]string, dockerImage string) {
-	for _, childNode := range jsonNode.Children {
-
-		if childNode.AttributesMap["jenkins.pipeline.step.type"] == "withDockerContainer" {
-			if img, ok := childNode.ParameterMap["image"].(string); ok {
-				dockerImage = img // Update Docker image from the node
-			}
+	if jsonNode.AttributesMap["jenkins.pipeline.step.type"] == "withDockerContainer" {
+		if img, ok := jsonNode.ParameterMap["image"].(string); ok {
+			dockerImage = img // Update Docker image from the node
 		}
+	}
 
-		if childNode.AttributesMap["jenkins.pipeline.step.type"] == "stage" {
+	//if _, ok := childNode.AttributesMap["jenkins.pipeline.step.id"]; ok {
+	if jsonNode.AttributesMap["jenkins.pipeline.step.type"] == "stage" || jsonNode.AttributesMap["jenkins.pipeline.step.type"] == "node" {
```

Comment:
I think the logic is that it recurse to find the `first` node or stage. `script` is not be possible at the root level. I was wrong in the call